### ISSUE TITLE
Can't manage users

### ISSUE THREAD

**Issue by Kyle9982** [cockpit-project/cockpit#13111]
Running Version 164 on Ubuntu 18.04
Console log:

Unchecked runtime.lastError: Could not establish connection. Receiving end does not exist.
cockpit/$584365dc6cfde824e94a531bbf132ebec97bbaff/users/index.html#/:1 Error handling response: TypeError: Cannot read property 'hiding' of undefined
    at Object.handleHiding (chrome-extension://bbmegnmpleoagolcnjnejdacakedpcgd/adblock/js/adblock_start_chrome.js:1:1539)
    at chrome-extension://bbmegnmpleoagolcnjnejdacakedpcgd/adblock/js/adblock_start_common.js:1:2541
users.js:290 Uncaught TypeError: Cannot read property 'match' of undefined
    at users.js:290
    at Array.filter (<anonymous>)
    at h.update (users.js:289)
    at Object.e (users.js:276)
    at l (cockpit.js:55)
    at k (cockpit.js:2659)
    at U.<anonymous> (cockpit.js:2557)
    at l (cockpit.js:55)
    at U.o (cockpit.js:162)
    at v (cockpit.js:521)
(anonymous) @ users.js:290
update @ users.js:289
e @ users.js:276
l @ cockpit.js:55
k @ cockpit.js:2659
(anonymous) @ cockpit.js:2557
l @ cockpit.js:55
o @ cockpit.js:162
v @ cockpit.js:521
m @ cockpit.js:530
S @ cockpit.js:429
M.i.onmessage.e.dispatch_data @ cockpit.js:364
t @ cockpit.js:226
postMessage (async)
(anonymous) @ index.js:3482
M.i.onmessage.e.dispatch_data @ cockpit.js:362

**Comment by Kyle9982** [comment]
![image](https://user-images.githubusercontent.com/42327407/68423971-e856b300-0157-11ea-8317-1bc0f47d393f.png)

**Comment by marusak** [comment]
I see in the log `adblock` and `chrome extension`. Can you please try to disable this extension (or ideally start in safe mode - meaning without any extensions)?
Does the problem persist?

**Comment by Kyle9982** [comment]
![image](https://user-images.githubusercontent.com/42327407/68974034-98ee3380-07a4-11ea-8a57-3d55a1ff2b4c.png)

Without any extentions

**Comment by martinpitt** [comment]
This sounds related to bug #13071. Can you please  check `getent passwd <username>` for the `<username>` that you are trying to login as? Based on the stack trace it can only really be this code:
```
        var accounts = this.accounts.filter(function(account) {
            return !((account.uid < 1000 && account.uid !== 0) ||
                account.shell.match(/^(\/usr)?\/sbin\/nologin/) ||
                account.shell === '/bin/false');
        });
```
and it's plausible that `account.shell` is undefined here.

This should be reproducible in a test case, I'll have a look.

**Comment by martinpitt** [comment]
With an empty shell field like

    test:x:1001:1002::/home/test:

the page does not crash. But I do get a similar crash when dropping the last `:`, i. e. the field isn't even present any more:

    test:x:1001:1002::/home/test
```
TypeError: account.shell is undefined users.js:35675:14
    accounts https://127.0.0.2:9091/cockpit/$a52aa19dcf723dbc555da8d4c6f71921ef4b593cc6f255609d5142f7a4415a93/users/users.js:35675
    filter self-hosted:269
    update https://127.0.0.2:9091/cockpit/$a52aa19dcf723dbc555da8d4c6f71921ef4b593cc6f255609d5142f7a4415a93/users/users.js:35674
    parse_accounts https://127.0.0.2:9091/cockpit/$a52aa19dcf723dbc555da8d4c6f71921ef4b593cc6f255609d5142f7a4415a93/users/users.js:35657
```

That's the only case that I can imagine and see in the code. I'll work on a fix.

**Comment by martinpitt** [comment]
Fixed in PR #13384
